libs/SmartSync/src/com/salesforce/androidsdk/smartsync/util/SmartSyncLogger.java

```diff
@@ -61,6 +70,20 @@ public static void e(String tag, String message, Throwable e) {
     }
 
     /**
+     * Logs an error log line.
+     *
+     * @param tag Log tag.
+     * @param origin Origin class.
+     * @param msg Log message.
+     * @param obj Object to be logged.
+     */
+    public static void e(String tag, Object origin, String msg, Object obj) {
```

Comment:
True. I can get rid of `origin` here.

And no, `SalesforceLogger` does not print component name in the console. The component name matters only for the file backed storage - each component has its own file. Also, depending on the log levels for different components, log lines printed will vary. However, the component names themselves wouldn't be a part of the log line. I don't know if there's much value in spewing those out to the console.